pytest-embedded-idf/pytest_embedded_idf/serial.py

```diff
@@ -84,30 +85,28 @@ def load_ram(self) -> None:
         if self.app.bin_file:
             bin_file = self.app.bin_file
         else:
-            live_print_call(
+            with contextlib.redirect_stdout(self._q):
+                esptool.main(
+                    [
+                        '--chip',
+                        self.app.target,
+                        'elf2image',
+                        self.app.elf_file,
+                        *self.app.write_flash_args,
+                    ]
+                )
+            bin_file = self.app.elf_file.replace('.elf', '.bin')
+
+        with contextlib.redirect_stdout(self._q):
+            esptool.main(
                 [
-                    'esptool.py',
                     '--chip',
                     self.app.target,
-                    'elf2image',
-                    self.app.elf_file,
-                    *self.app.write_flash_args,
+                    '--no-stub',
+                    'load_ram',
+                    bin_file,
                 ],
-                msg_queue=self._q,
             )
```

Comment:
esp=self.esp here?

otherwise the serial connection will be re-established.

Reply:
Yes, I think we can try adding `self.esp` to all `esptool.main` functions.